keyvi/src/cpp/dictionary/fsa/internal/json_value_store.h

```diff
@@ -292,22 +293,15 @@ class JsonValueStoreReader final: public IValueStoreReader {
 
     boost::interprocess::map_options_t map_options = boost::interprocess::default_map_options;
 
-#ifdef MAP_HUGETLB
-    map_options |= MAP_HUGETLB;
-#endif
-
-    if (!load_lazy) {
-#ifdef MAP_POPULATE
-      map_options |= MAP_POPULATE;
-#endif
-    }
-
     strings_region_ = new boost::interprocess::mapped_region(
         *file_mapping, boost::interprocess::read_only, offset,
         strings_size, 0, map_options);
 
-    // prevent pre-fetching pages by the OS which does not make sense as values usually fit into few pages
-    strings_region_->advise(boost::interprocess::mapped_region::advice_types::advice_random);
+    auto advise = internal::MemoryMapFlags::ValuesGetMemoryMapAdvices(loading_strategy);
+
+    if (advise != -1) {
```

Comment:
Can you please to explain this check ? 
As I understand you extended the existing `boost enum` by some invalid/default value.